app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,33 @@
+class AuthController < ApplicationController
+  def login
+    user = User.find_by(username: login_params[:username])
+    if user && user.authenticate(login_params[:password])
+      token = JWT.encode({ user_id: user.id }, Rails.application.secret_key_base, 'HS256')
+      render json: { user: user, token: token }
+    else
+      render json: {}, status: 401
```

Comment:
Does rails contain a definition of status codes anywhere? If so it could be good to use those. If not then I would define all of them in one place so you could access them by name.

Reply:
Rails does have these status codes. The references to 401 will be updated to `:unauthorized` in the next commit.